junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/DefaultLauncher.java

```diff
@@ -80,9 +80,12 @@ public TestPlan discover(LauncherDiscoveryRequest discoveryRequest) {
 	}
 
 	@Override
-	public void execute(LauncherDiscoveryRequest discoveryRequest) {
+	public void execute(LauncherDiscoveryRequest discoveryRequest, TestExecutionListener... listeners) {
 		Preconditions.notNull(discoveryRequest, "LauncherDiscoveryRequest must not be null");
+		Preconditions.notNull(listeners, "TestExecutionListener array must not be null");
+		this.listenerRegistry.registerListeners(listeners);
```

Comment:
Although we don't state that the `execute()` method can be launched in parallel from different threads for the same `Launcher` instance, it still makes me a bit uneasy to _mutate_ the listener registry in this manner.

I think I would prefer to introduce a composite wrapper in such cases and leave the original listener registry untouched.

@junit-team/junit-lambda, thoughts?